In [1]:

import torch
import gc
torch.cuda.empty_cache()
gc.collect()


0

In [2]:
import os
os.environ["TOKENIZERS_PARALLELISM"] = "false"
# Initialize the console

# Step 2: Import necessary libraries
from typing import Optional, Dict, Any
import os
import torch
import pytorch_lightning as pl
import pandas as pd
from transformers import AutoTokenizer
from torch.utils.data import Dataset, DataLoader
from rich.traceback import install
from pytorch_lightning.loggers import WandbLogger
import torch.nn as nn
import torch.optim as optim
from transformers import AutoConfig
from adapters import AutoAdapterModel, AdapterConfig
from adapters.composition import Stack
from pytorch_lightning import seed_everything, Trainer
from pytorch_lightning.callbacks import ModelCheckpoint,EarlyStopping
import torchmetrics

install(show_locals=True)

from setup import setup_src_path
print(setup_src_path())
import data.processed as processed
import config.config as config
import utils.setup as setup
import utils.functions as fn
from importlib import reload

from datasets import load_from_disk

print(config.Config.TXT_SAVE_PATH)
print(config.Config.MODEL_SAVE_PATH)

dataset = load_from_disk(f"../{config.Config.DATASETS_SAVE_PATH}/datasets")


['/usr/lib/python38.zip', '/usr/lib/python3.8', '/usr/lib/python3.8/lib-dynload', '', '/home/guest/.cache/pypoetry/virtualenvs/third-experments-xuKQSur9-py3.8/lib/python3.8/site-packages', '/tmp/tmpa20efblm', '/home/guest/Desktop/projects/third-experiments/domain_adaptation_project/modules']


2024-09-01 00:03:16.059400: I tensorflow/core/util/port.cc:110] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2024-09-01 00:03:16.090023: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F AVX512_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.


2024-09-01 00:03:16.657549: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT


./text-files/
./hp-model-


In [3]:
import torch
import os
import pytorch_lightning as pl
from transformers import  AutoConfig, DataCollatorForLanguageModeling
from collections import defaultdict
import torch.nn as nn
import torch.optim as optim
from torch.optim.lr_scheduler import ReduceLROnPlateau
import numpy as np
import torchmetrics

class JointDomainTaskAdapter(pl.LightningModule):
    def __init__(self, hparams,source_dataset_length,target_dataset_length):
        super(JointDomainTaskAdapter, self).__init__()
        self.save_hyperparameters(hparams)

        # Load config with hidden states output
        self.config = AutoConfig.from_pretrained(self.hparams["pretrained_model_name"])
        self.config.output_hidden_states = True
        self.model = AutoAdapterModel.from_pretrained(self.hparams["pretrained_model_name"], config=self.config)

        # Set reduction factor and leave_out layers
        self.reduction_factor = self.hparams.get("reduction_factor", 16)
        self.leave_out = self.hparams.get("leave_out", [])
        # if self.leave_out != "None":
        #     self.leave_out = self.leave_out.split(",")
        #     self.leave_out = [int(i) for i in self.leave_out]
        # else:
        #     self.leave_out = []

        # Load MLM adapter with head
        self.model.load_adapter(f"{self.hparams['saved_adapter_dir']}/{self.hparams['domain_adapter_name']}", with_head=True)

        # Add classification head for the task
        self.model.add_classification_head(f"{self.hparams['task_adapter_name']}", num_labels=self.hparams["num_classes"])

        # Set active adapters
        self.model.train_adapter(self.hparams['domain_adapter_name'])
        # Calculate alpha based on dataset lengths
        self.alpha = source_dataset_length / (source_dataset_length + target_dataset_length)

        # Initialize loss functions and metrics
        self.criterion = nn.CrossEntropyLoss()
        self.mlm_criterion = nn.CrossEntropyLoss()
        self.accuracy = torchmetrics.Accuracy(task='multiclass', num_classes=self.hparams["num_classes"])
        self.f1 = torchmetrics.F1Score(task='multiclass', num_classes=self.hparams["num_classes"], average="weighted")
        self.f1_macro = torchmetrics.F1Score(task='multiclass', num_classes=self.hparams["num_classes"], average="macro")
        self.softmax = nn.Softmax(dim=1)
        self.validation_outputs = []
        self.test_outputs = []
        # Optimizer related variables
        self.learning_rate = self.hparams.get("learning_rate", 1e-4)
        self.scheduler_factor = self.hparams.get("scheduler_factor", 0.1)
        self.scheduler_patience = self.hparams.get("scheduler_patience", 0.05)
        self.scheduler_threshold = self.hparams.get("scheduler_threshold", 0.0001)
        self.scheduler_cooldown = self.hparams.get("scheduler_cooldown", 0)
        self.scheduler_eps = self.hparams.get("scheduler_eps", 1e-8)

    def forward(self, input_ids, attention_mask=None, labels=None, task=None):
        if task == "mlm":
            self.model.active_head= self.hparams['domain_adapter_name']
            outputs = self.model(input_ids=input_ids, attention_mask=attention_mask, labels=labels)
        elif task == "classification":
            self.model.active_head= self.hparams['task_adapter_name']
            outputs = self.model(input_ids=input_ids, attention_mask=attention_mask, labels=labels)
        else:
            raise ValueError("Task must be either 'mlm' or 'classification'.")
        return outputs

    def training_step(self, batch, batch_idx):
        # Separate source and target data processing
        source_input_ids = batch["source_input_ids"]
        source_attention_mask = batch["source_attention_mask"]
        source_labels = batch["label_source"]

        target_input_ids = batch["target_input_ids"]
        target_attention_mask = batch["target_attention_mask"]
        mlm_labels = batch["mlm_labels"]

        # Calculate dynamic alpha based on the lengths of source and target data
        alpha = self.alpha

        # Classification task
        cls_outputs = self(input_ids=source_input_ids, attention_mask=source_attention_mask, task="classification")
        cls_logits = cls_outputs.logits
        task_loss = self.criterion(cls_logits, source_labels)

        # MLM task
        mlm_outputs = self(input_ids=target_input_ids, attention_mask=target_attention_mask, labels=mlm_labels, task="mlm")
        mlm_loss = mlm_outputs.loss

        # Combine losses
        loss = alpha * task_loss + (1 - alpha) * mlm_loss

        accuracy = self.accuracy(source_labels, torch.argmax(self.softmax(cls_logits), dim=1))
        f1 = self.f1(source_labels, torch.argmax(self.softmax(cls_logits), dim=1))

        metrics = {
            "train/accuracy": accuracy,
            "train/f1": f1,
            "train/taskclf_loss": task_loss,
            "train/loss": loss,
            "train/mlm_loss": mlm_loss
        }

        for key, val in metrics.items():
            self.log(name=key, value=val)

        return loss

    def validation_step(self, batch, batch_idx):
        # Separate source and target data processing
        source_input_ids = batch["source_input_ids"]
        source_attention_mask = batch["source_attention_mask"]
        source_labels = batch["label_source"]

        target_input_ids = batch["target_input_ids"]
        target_attention_mask = batch["target_attention_mask"]
        mlm_labels = batch["mlm_labels"]

        # Calculate dynamic alpha based on the lengths of source and target data
        alpha = self.alpha
        # Classification task
        cls_outputs = self(input_ids=source_input_ids, attention_mask=source_attention_mask, task="classification")
        cls_logits = cls_outputs.logits
        task_loss = self.criterion(cls_logits, source_labels)

        # MLM task
        mlm_outputs = self(input_ids=target_input_ids, attention_mask=target_attention_mask, labels=mlm_labels, task="mlm")
        mlm_loss = mlm_outputs.loss

        # Combine losses
        loss = alpha * task_loss + (1 - alpha) * mlm_loss

        accuracy = self.accuracy(source_labels, torch.argmax(self.softmax(cls_logits), dim=1))
        f1 = self.f1(source_labels, torch.argmax(self.softmax(cls_logits), dim=1))
        self.validation_outputs.append({
            "val/accuracy": accuracy,
            "val/f1": f1,
            "val/taskclf_loss": task_loss,
            "val/loss": loss,
            "val/mlm_loss": mlm_loss
                })
        
        metrics = {
            "val/accuracy": accuracy,
            "val/f1": f1,
            "val/taskclf_loss": task_loss,
            "val/loss": loss,
            "val/mlm_loss": mlm_loss
        }

        for key, val in metrics.items():
            self.log(name=key, value=val)

        return metrics
    def on_validation_epoch_start(self):
        self.validation_outputs = []
    
    def on_validation_epoch_end(self):
        outputs= self.validation_outputs

        avg_loss = torch.stack([x["val/loss"] for x in outputs]).mean()
        avg_task_loss = torch.stack([x["val/taskclf_loss"] for x in outputs]).mean()
        avg_mlm_loss = torch.stack([x["val/mlm_loss"] for x in outputs]).mean()
        avg_accuracy = torch.stack([x["val/accuracy"] for x in outputs]).mean()
        avg_f1 = torch.stack([x["val/f1"] for x in outputs]).mean()
        print(f"val/accuracy: {avg_accuracy}")
        print(f"val/f1: {avg_f1}")
        print(f"val/taskclf_loss: {avg_task_loss}")
        print(f"val/loss: {avg_loss}")
        print(f"val/mlm_loss: {avg_mlm_loss}")
        metrics = {
            "val/avg_loss": avg_loss,
            "val/avg_taskclf_loss": avg_task_loss,
            "val/avg_mlm_loss": avg_mlm_loss,
            "val/avg_accuracy": avg_accuracy,
            "val/avg_f1": avg_f1,
        }

        for key, val in metrics.items():
            self.log(name=key, value=val)
        self.log("val_loss", avg_loss)

    def test_step(self, batch, batch_idx):
        # Separate source and target data processing
        source_input_ids = batch["source_input_ids"]
        source_attention_mask = batch["source_attention_mask"]
        source_labels = batch["label_source"]

        target_input_ids = batch["target_input_ids"]
        target_attention_mask = batch["target_attention_mask"]
        target_labels = batch["label_target"]

        # Classification task for source data
        cls_outputs_source = self(input_ids=source_input_ids, attention_mask=source_attention_mask, task="classification")
        cls_logits_source = cls_outputs_source.logits
        task_loss_source = self.criterion(cls_logits_source, source_labels)

        # Classification task for target data
        cls_outputs_target = self(input_ids=target_input_ids, attention_mask=target_attention_mask, task="classification")
        cls_logits_target = cls_outputs_target.logits
        task_loss_target = self.criterion(cls_logits_target, target_labels)

        # Combine losses (though typically you would evaluate them separately)
        loss = task_loss_source + task_loss_target

        accuracy_source = self.accuracy(source_labels, torch.argmax(self.softmax(cls_logits_source), dim=1))
        f1_source = self.f1(source_labels, torch.argmax(self.softmax(cls_logits_source), dim=1))
        f1_macro_source = self.f1_macro(source_labels, torch.argmax(self.softmax(cls_logits_source), dim=1))

        accuracy_target = self.accuracy(target_labels, torch.argmax(self.softmax(cls_logits_target), dim=1))
        f1_target = self.f1(target_labels, torch.argmax(self.softmax(cls_logits_target), dim=1))
        f1_macro_target = self.f1_macro(target_labels, torch.argmax(self.softmax(cls_logits_target), dim=1))

        metrics = {
            "source_test/loss": task_loss_source,
            "source_test/accuracy": accuracy_source,
            "source_test/f1": f1_source,
            "source_test/f1_macro": f1_macro_source,
            "target_test/loss": task_loss_target,
            "target_test/accuracy": accuracy_target,
            "target_test/f1": f1_target,
            "target_test/f1_macro": f1_macro_target,
        }

        for key, val in metrics.items():
            self.log(name=key, value=val)

        self.test_outputs.append({
            "source_test/loss": task_loss_source,
            "source_test/accuracy": accuracy_source,
            "source_test/f1": f1_source,
            "source_test/f1_macro": f1_macro_source,
            "target_test/loss": task_loss_target,
            "target_test/accuracy": accuracy_target,
            "target_test/f1": f1_target,
            "target_test/f1_macro": f1_macro_target,
        })
        return metrics
    def on_test_epoch_start(self):
        self.test_outputs = []

    def on_test_epoch_end(self):
        outputs=  self.test_outputs

        avg_loss_source = torch.stack([x["source_test/loss"] for x in outputs]).mean()
        avg_task_loss_target = torch.stack([x["target_test/loss"] for x in outputs]).mean()
        avg_accuracy_source = torch.stack([x["source_test/accuracy"] for x in outputs]).mean()
        avg_f1_source = torch.stack([x["source_test/f1"] for x in outputs]).mean()
        avg_f1_macro_source = torch.stack([x["source_test/f1_macro"] for x in outputs]).mean()

        avg_accuracy_target = torch.stack([x["target_test/accuracy"] for x in outputs]).mean()
        avg_f1_target = torch.stack([x["target_test/f1"] for x in outputs]).mean()
        avg_f1_macro_target = torch.stack([x["target_test/f1_macro"] for x in outputs]).mean()

        metrics = {
            "source_test/loss": avg_loss_source,
            "target_test/loss": avg_task_loss_target,
            "source_test/accuracy": avg_accuracy_source,
            "source_test/f1": avg_f1_source,
            "source_test/f1_macro": avg_f1_macro_source,
            "target_test/accuracy": avg_accuracy_target,
            "target_test/f1": avg_f1_target,
            "target_test/f1_macro": avg_f1_macro_target,
        }

        for key, val in metrics.items():
            self.log(name=key, value=val)
    def save_adapter(self, location, adapter_name):
        self.model.save_adapter(location, adapter_name)

    def configure_optimizers(self):
        optimizer = optim.AdamW(self.parameters(), lr=self.learning_rate)
        lr_scheduler = ReduceLROnPlateau(
            optimizer=optimizer,
            mode="min",
            factor=self.scheduler_factor,
            patience=self.scheduler_patience,
            threshold=self.scheduler_threshold,
            cooldown=self.scheduler_cooldown,
            eps=self.scheduler_eps,
            verbose=True,
        )
        return [optimizer], [{"scheduler": lr_scheduler, "reduce_lr_on_plateau": True, "monitor": "val_loss", "interval": "epoch"}]


In [4]:
import wandb

wandb.login()
# Wandb setup and training loop
seeds = [42, 10, 100]  # List of seeds
project_name = 'mixed_edited'  # Replace with your wandb project name
domain = 'TES'  # Replace with the specific domain for this notebook
type = 'union'  # Replace with the specific type for this notebook

# Initialize results dictionary
results = {
    "last_epoch": {
        "source_test/loss": [],
        "source_test/accuracy": [],
        "source_test/f1": [],
        "source_test/f1_macro": [],
        "target_test/loss": [],
        "target_test/accuracy": [],
        "target_test/f1": [],
        "target_test/f1_macro": [],
    },
    "best_model": {
        "source_test/loss": [],
        "source_test/accuracy": [],
        "source_test/f1": [],
        "source_test/f1_macro": [],
        "target_test/loss": [],
        "target_test/accuracy": [],
        "target_test/f1": [],
        "target_test/f1_macro": [],
    },
    "epoch_saved": {
        "source_test/loss": [],
        "source_test/accuracy": [],
        "source_test/f1": [],
        "source_test/f1_macro": [],
        "target_test/loss": [],
        "target_test/accuracy": [],
        "target_test/f1": [],
        "target_test/f1_macro": [],
    }
}

best_val_loss = float('inf')
best_model = None
best_model_path = ""

wandb: Currently logged in as: mrawhani5 (mrawhani). Use `wandb login --relogin` to force relogin


In [5]:
reload(processed)
for seed in seeds:
    #wandb.init(project=project_name, name=f'{domain}_{type}_run_with_seed_{seed}', config={'seed': seed})

    try:
        seed_everything(seed)

        hparams = {
            "source_target": "telephone_slate",
            "source_domain": "telephone",
            "target_domain": "slate",
            "domain_adapter_name": "mlm_union_S",
            "task_adapter_name": "task_TESUni",
            "pretrained_model_name": "bert-base-uncased",
            "padding": "max_length",
            "max_seq_length": 128,
            "bsz": 32,
            "num_classes": 3,
            "learning_rate": 1e-4,
            "reduction_factor": 16,
            "mode": "domain",
            "saved_adapter_dir": "../../saved/adapters",
        }

        save_dir = "checkpoints"
        save_epoch_3 = 6  # Save model at the 3rd epoch
        
        dm = processed.DataModuleSourceTargetMixed(hparams)
        dm.setup('fit')
        dm.setup("test")
        source_length, target_length = dm.get_dataset_lengths()
        print(f"Source dataset length: {source_length}")
        print(f"Target dataset length: {target_length}")
        model = JointDomainTaskAdapter(hparams,source_length,target_length)

        checkpoint_callback = ModelCheckpoint(
            filename="task-TESUni-{epoch:02d}-{val_loss:.2f}",
            save_top_k=1,
            monitor="val_loss",
            mode="min",
        )
        save_model_callback_epoch = ModelCheckpoint(
            filename="TESUni-{epoch:02d}",
            every_n_epochs=save_epoch_3,
            save_top_k=-1,
        )

        #wandb_logger = WandbLogger()

    except Exception as e:
        print(f"Error during preprocessing : {e}")

    try:
        train_loader = dm.train_dataloader()
        val_loader = dm.val_dataloader()
        trainer = Trainer(
            max_epochs=10,
            accelerator="auto",
            precision=16,
            
            default_root_dir="checkpoints",
            #logger=wandb_logger,
            callbacks=[checkpoint_callback, save_model_callback_epoch],
            limit_train_batches=1.0,
            limit_val_batches=1.0,
            limit_test_batches=1.0,
        )

        trainer.fit(model, train_loader, val_loader)
        print(f"Best checkpoint path: {checkpoint_callback.best_model_path}")
        print(f"Saved epoch checkpoint path: {save_model_callback_epoch.best_model_path}")
    except Exception as e:
        print(f"Error during training : {e}")

    try:
        dm.setup("test")
        test_loader = dm.test_dataloader()
        test_results_last = trainer.test(model, test_loader)
        print("Test Results Last Epoch:", test_results_last)

        # Collect results for last epoch model
        for key, value in test_results_last[0].items():
            results["last_epoch"][key].append(value)

        # Paths to the saved checkpoints
        best_checkpoint_path = checkpoint_callback.best_model_path
        saved_epoch_checkpoint_path = save_model_callback_epoch.best_model_path
        print(f"Best checkpoint path: {best_checkpoint_path}")
        print(f"Saved epoch checkpoint path: {saved_epoch_checkpoint_path}")

        best_model = JointDomainTaskAdapter.load_from_checkpoint(best_checkpoint_path,source_dataset_length=source_length, target_dataset_length=target_length)
        test_results_best = trainer.test(best_model, test_loader)
        print("Test Results on Best Model:", test_results_best)
        for key, value in test_results_best[0].items():
            results["best_model"][key].append(value)

        saved_epoch_model = JointDomainTaskAdapter.load_from_checkpoint(saved_epoch_checkpoint_path,source_dataset_length=source_length, target_dataset_length=target_length)
        test_results_saved_epoch = trainer.test(saved_epoch_model, test_loader)
        print("Test Results on saved epoch:", test_results_saved_epoch)
        for key, value in test_results_saved_epoch[0].items():
            results["epoch_saved"][key].append(value)

    except Exception as e:
        print(f"Error during testing: {e}")

    #wandb.finish()

Seed set to 42


/home/guest/.cache/pypoetry/virtualenvs/third-experments-xuKQSur9-py3.8/lib/python3.8/site-packages/huggingface_hub/file_download.py:1132: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.
  warnings.warn(


Batch size: 32


Source genre: telephone
Target genre: slate
Number of target samples: 69575


Source genre: telephone


Target genre: slate
Number of target samples: 69575
Source dataset length: 75013
Target dataset length: 21585


Some weights of BertAdapterModel were not initialized from the model checkpoint at bert-base-uncased and are newly initialized: ['heads.default.3.bias']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.


/home/guest/.cache/pypoetry/virtualenvs/third-experments-xuKQSur9-py3.8/lib/python3.8/site-packages/torch/utils/data/dataloader.py:558: UserWarning: This DataLoader will create 16 worker processes in total. Our suggested max number of worker in current system is 12, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  warnings.warn(_create_warning_msg(
/home/guest/.cache/pypoetry/virtualenvs/third-experments-xuKQSur9-py3.8/lib/python3.8/site-packages/lightning_fabric/connector.py:563: `precision=16` is supported for historical reasons but its usage is discouraged. Please set your precision to 16-mixed instead!
Using 16bit Automatic Mixed Precision (AMP)


GPU available: True (cuda), used: True


TPU available: False, using: 0 TPU cores


IPU available: False, using: 0 IPUs


HPU available: False, using: 0 HPUs


`Trainer(limit_train_batches=1.0)` was configured so 100% of the batches per epoch will be used..


`Trainer(limit_val_batches=1.0)` was configured so 100% of the batches will be used..


`Trainer(limit_test_batches=1.0)` was configured so 100% of the batches will be used..


Missing logger folder: checkpoints/lightning_logs


LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]


eee


/home/guest/.cache/pypoetry/virtualenvs/third-experments-xuKQSur9-py3.8/lib/python3.8/site-packages/pytorch_lightning/core/optimizer.py:257: Found unsupported keys in the lr scheduler dict: {'reduce_lr_on_plateau'}. HINT: remove them from the output of `configure_optimizers`.

  | Name          | Type               | Params
-----------------------------------------------------
0 | model         | BertAdapterModel   | 119 M 
1 | criterion     | CrossEntropyLoss   | 0     
2 | mlm_criterion | CrossEntropyLoss   | 0     
3 | accuracy      | MulticlassAccuracy | 0     
4 | f1            | MulticlassF1Score  | 0     
5 | f1_macro      | MulticlassF1Score  | 0     
6 | softmax       | Softmax            | 0     
-----------------------------------------------------
9.5 M     Trainable params
109 M     Non-trainable params
119 M     Total params
476.013   Total estimated model params size (MB)


Sanity Checking: |                                                                                            …

val/accuracy: 0.328125
val/f1: 0.46674108505249023
val/taskclf_loss: 1.106689453125
val/loss: 1.284916639328003
val/mlm_loss: 1.9042987823486328


Training: |                                                                                                   …

Validation: |                                                                                                 …

val/accuracy: 0.7678799629211426
val/f1: 0.7687401175498962
val/taskclf_loss: 0.5603954195976257
val/loss: 0.8338254690170288
val/mlm_loss: 1.7840598821640015


Validation: |                                                                                                 …

val/accuracy: 0.7791507244110107
val/f1: 0.7797816395759583
val/taskclf_loss: 0.542116641998291
val/loss: 0.8219900727272034
val/mlm_loss: 1.7946168184280396


Validation: |                                                                                                 …

val/accuracy: 0.7813058495521545
val/f1: 0.7824388742446899
val/taskclf_loss: 0.5601878762245178
val/loss: 0.8340222835540771
val/mlm_loss: 1.7856619358062744


Validation: |                                                                                                 …

val/accuracy: 0.7936223149299622
val/f1: 0.7939064502716064
val/taskclf_loss: 0.6001781821250916
val/loss: 0.8598576188087463
val/mlm_loss: 1.7623056173324585


Validation: |                                                                                                 …

val/accuracy: 0.7959291338920593
val/f1: 0.7969070076942444
val/taskclf_loss: 0.5770742893218994
val/loss: 0.8401689529418945
val/mlm_loss: 1.7544854879379272


Validation: |                                                                                                 …

val/accuracy: 0.7972461581230164
val/f1: 0.7981223464012146
val/taskclf_loss: 0.5748772621154785
val/loss: 0.841346800327301
val/mlm_loss: 1.7673918008804321


Validation: |                                                                                                 …

val/accuracy: 0.7972461581230164
val/f1: 0.798103928565979
val/taskclf_loss: 0.5746821165084839
val/loss: 0.8369223475456238
val/mlm_loss: 1.7482695579528809


Validation: |                                                                                                 …

val/accuracy: 0.7968869805335999
val/f1: 0.7977375984191895
val/taskclf_loss: 0.5744446516036987
val/loss: 0.8408291935920715
val/mlm_loss: 1.7665786743164062


Validation: |                                                                                                 …

val/accuracy: 0.7967672348022461
val/f1: 0.7976281046867371
val/taskclf_loss: 0.574231743812561
val/loss: 0.8423925638198853
val/mlm_loss: 1.7743151187896729


Validation: |                                                                                                 …

`Trainer.fit` stopped: `max_epochs=10` reached.


val/accuracy: 0.7968869805335999
val/f1: 0.79774409532547
val/taskclf_loss: 0.5740534067153931
val/loss: 0.8411875367164612
val/mlm_loss: 1.7695424556732178


Best checkpoint path: checkpoints/lightning_logs/version_0/checkpoints/task-TESUni-epoch=01-val_loss=0.82.ckpt
Saved epoch checkpoint path: checkpoints/lightning_logs/version_0/checkpoints/TESUni-epoch=05.ckpt


LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]


Source genre: telephone
Target genre: slate
Number of target samples: 69575


Testing: |                                                                                                    …

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Test metric        ┃       DataLoader 0        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│   source_test/accuracy    │    0.8117079138755798     │
│      source_test/f1       │    0.8116697072982788     │
│   source_test/f1_macro    │    0.8027613162994385     │
│     source_test/loss      │     0.573732852935791     │
│   target_test/accuracy    │    0.7332949042320251     │
│      target_test/f1       │    0.7329106330871582     │
│   target_test/f1_macro    │     0.723328709602356     │
│     target_test/loss      │     0.759901762008667     │
└───────────────────────────┴───────────────────────────┘

Test Results Last Epoch: [{'source_test/loss': 0.573732852935791, 'source_test/accuracy': 0.8117079138755798, 'source_test/f1': 0.8116697072982788, 'source_test/f1_macro': 0.8027613162994385, 'target_test/loss': 0.759901762008667, 'target_test/accuracy': 0.7332949042320251, 'target_test/f1': 0.7329106330871582, 'target_test/f1_macro': 0.723328709602356}]
Best checkpoint path: checkpoints/lightning_logs/version_0/checkpoints/task-TESUni-epoch=01-val_loss=0.82.ckpt
Saved epoch checkpoint path: checkpoints/lightning_logs/version_0/checkpoints/TESUni-epoch=05.ckpt


Some weights of BertAdapterModel were not initialized from the model checkpoint at bert-base-uncased and are newly initialized: ['heads.default.3.bias']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.


LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]


Testing: |                                                                                                    …

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Test metric        ┃       DataLoader 0        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│   source_test/accuracy    │    0.7986031174659729     │
│      source_test/f1       │    0.7992354035377502     │
│   source_test/f1_macro    │    0.7878866791725159     │
│     source_test/loss      │    0.5176754593849182     │
│   target_test/accuracy    │     0.700532853603363     │
│      target_test/f1       │    0.6990686655044556     │
│   target_test/f1_macro    │    0.6926193833351135     │
│     target_test/loss      │    0.7109049558639526     │
└───────────────────────────┴───────────────────────────┘

Test Results on Best Model: [{'source_test/loss': 0.5176754593849182, 'source_test/accuracy': 0.7986031174659729, 'source_test/f1': 0.7992354035377502, 'source_test/f1_macro': 0.7878866791725159, 'target_test/loss': 0.7109049558639526, 'target_test/accuracy': 0.700532853603363, 'target_test/f1': 0.6990686655044556, 'target_test/f1_macro': 0.6926193833351135}]


Some weights of BertAdapterModel were not initialized from the model checkpoint at bert-base-uncased and are newly initialized: ['heads.default.3.bias']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.


LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]


Testing: |                                                                                                    …

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Test metric        ┃       DataLoader 0        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│   source_test/accuracy    │    0.8122119903564453     │
│      source_test/f1       │    0.8121238350868225     │
│   source_test/f1_macro    │    0.8033238649368286     │
│     source_test/loss      │    0.5743333101272583     │
│   target_test/accuracy    │    0.7327908873558044     │
│      target_test/f1       │    0.7324329018592834     │
│   target_test/f1_macro    │    0.7226143479347229     │
│     target_test/loss      │     0.760591447353363     │
└───────────────────────────┴───────────────────────────┘

Seed set to 10


Test Results on saved epoch: [{'source_test/loss': 0.5743333101272583, 'source_test/accuracy': 0.8122119903564453, 'source_test/f1': 0.8121238350868225, 'source_test/f1_macro': 0.8033238649368286, 'target_test/loss': 0.760591447353363, 'target_test/accuracy': 0.7327908873558044, 'target_test/f1': 0.7324329018592834, 'target_test/f1_macro': 0.7226143479347229}]
Batch size: 32


Source genre: telephone


Target genre: slate
Number of target samples: 69575


Source genre: telephone
Target genre: slate
Number of target samples: 69575
Source dataset length: 75013
Target dataset length: 21585


Some weights of BertAdapterModel were not initialized from the model checkpoint at bert-base-uncased and are newly initialized: ['heads.default.3.bias']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.


Using 16bit Automatic Mixed Precision (AMP)


GPU available: True (cuda), used: True


TPU available: False, using: 0 TPU cores


IPU available: False, using: 0 IPUs


HPU available: False, using: 0 HPUs


`Trainer(limit_train_batches=1.0)` was configured so 100% of the batches per epoch will be used..


`Trainer(limit_val_batches=1.0)` was configured so 100% of the batches will be used..


`Trainer(limit_test_batches=1.0)` was configured so 100% of the batches will be used..


LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]



  | Name          | Type               | Params
-----------------------------------------------------
0 | model         | BertAdapterModel   | 119 M 
1 | criterion     | CrossEntropyLoss   | 0     
2 | mlm_criterion | CrossEntropyLoss   | 0     
3 | accuracy      | MulticlassAccuracy | 0     
4 | f1            | MulticlassF1Score  | 0     
5 | f1_macro      | MulticlassF1Score  | 0     
6 | softmax       | Softmax            | 0     
-----------------------------------------------------
9.5 M     Trainable params
109 M     Non-trainable params
119 M     Total params
476.013   Total estimated model params size (MB)


eee


Sanity Checking: |                                                                                            …

val/accuracy: 0.234375
val/f1: 0.25993409752845764
val/taskclf_loss: 1.109283447265625
val/loss: 1.2556959390640259
val/mlm_loss: 1.7645139694213867


Training: |                                                                                                   …

Validation: |                                                                                                 …

val/accuracy: 0.7714559435844421
val/f1: 0.7718485593795776
val/taskclf_loss: 0.557565450668335
val/loss: 0.8328756093978882
val/mlm_loss: 1.7896440029144287


Validation: |                                                                                                 …

val/accuracy: 0.7836845517158508
val/f1: 0.7833713293075562
val/taskclf_loss: 0.5341743230819702
val/loss: 0.8132825493812561
val/mlm_loss: 1.783250331878662


Validation: |                                                                                                 …

val/accuracy: 0.789311945438385
val/f1: 0.7891321182250977
val/taskclf_loss: 0.5501735806465149
val/loss: 0.8294327855110168
val/mlm_loss: 1.7999250888824463


Validation: |                                                                                                 …

val/accuracy: 0.7966474890708923
val/f1: 0.7967719435691833
val/taskclf_loss: 0.5899022817611694
val/loss: 0.8553076386451721
val/mlm_loss: 1.7776544094085693


Validation: |                                                                                                 …

val/accuracy: 0.7974856495857239
val/f1: 0.797954797744751
val/taskclf_loss: 0.5750665664672852
val/loss: 0.8398424983024597
val/mlm_loss: 1.7600017786026


Validation: |                                                                                                 …

val/accuracy: 0.7977250814437866
val/f1: 0.7981227040290833
val/taskclf_loss: 0.573478102684021
val/loss: 0.8395905494689941
val/mlm_loss: 1.7643942832946777


Validation: |                                                                                                 …

val/accuracy: 0.7977250814437866
val/f1: 0.7981227040290833
val/taskclf_loss: 0.5732834339141846
val/loss: 0.8389483690261841
val/mlm_loss: 1.7621972560882568


Validation: |                                                                                                 …

val/accuracy: 0.7976053357124329
val/f1: 0.7980199456214905
val/taskclf_loss: 0.5730866193771362
val/loss: 0.8407655954360962
val/mlm_loss: 1.7710134983062744


Validation: |                                                                                                 …

val/accuracy: 0.7978448271751404
val/f1: 0.7982740998268127
val/taskclf_loss: 0.5728518962860107
val/loss: 0.8420065641403198
val/mlm_loss: 1.7773829698562622


Validation: |                                                                                                 …

`Trainer.fit` stopped: `max_epochs=10` reached.


val/accuracy: 0.7977250814437866
val/f1: 0.7981577515602112
val/taskclf_loss: 0.5726881623268127
val/loss: 0.8381413221359253
val/mlm_loss: 1.7606539726257324


Best checkpoint path: checkpoints/lightning_logs/version_1/checkpoints/task-TESUni-epoch=01-val_loss=0.81.ckpt
Saved epoch checkpoint path: checkpoints/lightning_logs/version_1/checkpoints/TESUni-epoch=05.ckpt


Source genre: telephone
Target genre: slate
Number of target samples: 69575


LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]


Testing: |                                                                                                    …

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Test metric        ┃       DataLoader 0        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│   source_test/accuracy    │     0.822796642780304     │
│      source_test/f1       │    0.8225594162940979     │
│   source_test/f1_macro    │    0.8125101327896118     │
│     source_test/loss      │    0.5562842488288879     │
│   target_test/accuracy    │    0.7282546162605286     │
│      target_test/f1       │    0.7278156876564026     │
│   target_test/f1_macro    │     0.717542290687561     │
│     target_test/loss      │    0.7413967847824097     │
└───────────────────────────┴───────────────────────────┘

Test Results Last Epoch: [{'source_test/loss': 0.5562842488288879, 'source_test/accuracy': 0.822796642780304, 'source_test/f1': 0.8225594162940979, 'source_test/f1_macro': 0.8125101327896118, 'target_test/loss': 0.7413967847824097, 'target_test/accuracy': 0.7282546162605286, 'target_test/f1': 0.7278156876564026, 'target_test/f1_macro': 0.717542290687561}]
Best checkpoint path: checkpoints/lightning_logs/version_1/checkpoints/task-TESUni-epoch=01-val_loss=0.81.ckpt
Saved epoch checkpoint path: checkpoints/lightning_logs/version_1/checkpoints/TESUni-epoch=05.ckpt


Some weights of BertAdapterModel were not initialized from the model checkpoint at bert-base-uncased and are newly initialized: ['heads.default.3.bias']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.


LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]


Testing: |                                                                                                    …

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Test metric        ┃       DataLoader 0        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│   source_test/accuracy    │    0.7967309951782227     │
│      source_test/f1       │    0.7970972061157227     │
│   source_test/f1_macro    │    0.7871140837669373     │
│     source_test/loss      │    0.5136334896087646     │
│   target_test/accuracy    │    0.7217021584510803     │
│      target_test/f1       │    0.7209112644195557     │
│   target_test/f1_macro    │    0.7127382755279541     │
│     target_test/loss      │    0.6852752566337585     │
└───────────────────────────┴───────────────────────────┘

Test Results on Best Model: [{'source_test/loss': 0.5136334896087646, 'source_test/accuracy': 0.7967309951782227, 'source_test/f1': 0.7970972061157227, 'source_test/f1_macro': 0.7871140837669373, 'target_test/loss': 0.6852752566337585, 'target_test/accuracy': 0.7217021584510803, 'target_test/f1': 0.7209112644195557, 'target_test/f1_macro': 0.7127382755279541}]


Some weights of BertAdapterModel were not initialized from the model checkpoint at bert-base-uncased and are newly initialized: ['heads.default.3.bias']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.


LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]


Testing: |                                                                                                    …

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Test metric        ┃       DataLoader 0        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│   source_test/accuracy    │    0.8233006596565247     │
│      source_test/f1       │    0.8231325149536133     │
│   source_test/f1_macro    │    0.8129770755767822     │
│     source_test/loss      │    0.5568398237228394     │
│   target_test/accuracy    │    0.7287586331367493     │
│      target_test/f1       │     0.728269100189209     │
│   target_test/f1_macro    │     0.718044638633728     │
│     target_test/loss      │    0.7423831224441528     │
└───────────────────────────┴───────────────────────────┘

Seed set to 100


Test Results on saved epoch: [{'source_test/loss': 0.5568398237228394, 'source_test/accuracy': 0.8233006596565247, 'source_test/f1': 0.8231325149536133, 'source_test/f1_macro': 0.8129770755767822, 'target_test/loss': 0.7423831224441528, 'target_test/accuracy': 0.7287586331367493, 'target_test/f1': 0.728269100189209, 'target_test/f1_macro': 0.718044638633728}]
Batch size: 32


Source genre: telephone
Target genre: slate
Number of target samples: 69575


Source genre: telephone
Target genre: slate
Number of target samples: 69575


Source dataset length: 75013
Target dataset length: 21585


Some weights of BertAdapterModel were not initialized from the model checkpoint at bert-base-uncased and are newly initialized: ['heads.default.3.bias']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.


Using 16bit Automatic Mixed Precision (AMP)


GPU available: True (cuda), used: True


TPU available: False, using: 0 TPU cores


IPU available: False, using: 0 IPUs


HPU available: False, using: 0 HPUs


`Trainer(limit_train_batches=1.0)` was configured so 100% of the batches per epoch will be used..


`Trainer(limit_val_batches=1.0)` was configured so 100% of the batches will be used..


`Trainer(limit_test_batches=1.0)` was configured so 100% of the batches will be used..


LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]



  | Name          | Type               | Params
-----------------------------------------------------
0 | model         | BertAdapterModel   | 119 M 
1 | criterion     | CrossEntropyLoss   | 0     
2 | mlm_criterion | CrossEntropyLoss   | 0     
3 | accuracy      | MulticlassAccuracy | 0     
4 | f1            | MulticlassF1Score  | 0     
5 | f1_macro      | MulticlassF1Score  | 0     
6 | softmax       | Softmax            | 0     
-----------------------------------------------------
9.5 M     Trainable params
109 M     Non-trainable params
119 M     Total params
476.013   Total estimated model params size (MB)


eee


Sanity Checking: |                                                                                            …

val/accuracy: 0.390625
val/f1: 0.5528455376625061
val/taskclf_loss: 1.08837890625
val/loss: 1.2304290533065796
val/mlm_loss: 1.7240869998931885


Training: |                                                                                                   …

Validation: |                                                                                                 …

val/accuracy: 0.7717113494873047
val/f1: 0.7713525295257568
val/taskclf_loss: 0.5630925893783569
val/loss: 0.8350611925125122
val/mlm_loss: 1.7802165746688843


Validation: |                                                                                                 …

val/accuracy: 0.7872605323791504
val/f1: 0.7876025438308716
val/taskclf_loss: 0.5365691781044006
val/loss: 0.8178388476371765
val/mlm_loss: 1.7953180074691772


Validation: |                                                                                                 …

val/accuracy: 0.7917225360870361
val/f1: 0.7922080159187317
val/taskclf_loss: 0.5372147560119629
val/loss: 0.8161357641220093
val/mlm_loss: 1.7854522466659546


Validation: |                                                                                                 …

val/accuracy: 0.7901819944381714
val/f1: 0.7905999422073364
val/taskclf_loss: 0.596862256526947
val/loss: 0.8663314580917358
val/mlm_loss: 1.8028010129928589


Validation: |                                                                                                 …

val/accuracy: 0.7978448271751404
val/f1: 0.797971248626709
val/taskclf_loss: 0.6476531624794006
val/loss: 0.8972642421722412
val/mlm_loss: 1.7647221088409424


Validation: |                                                                                                 …

val/accuracy: 0.7976053357124329
val/f1: 0.7977798581123352
val/taskclf_loss: 0.6252181529998779
val/loss: 0.8841760754585266
val/mlm_loss: 1.7841161489486694


Validation: |                                                                                                 …

val/accuracy: 0.7978448271751404
val/f1: 0.7980220317840576
val/taskclf_loss: 0.6231735348701477
val/loss: 0.8794103264808655
val/mlm_loss: 1.7698943614959717


Validation: |                                                                                                 …

val/accuracy: 0.7977250814437866
val/f1: 0.7979081273078918
val/taskclf_loss: 0.6229500770568848
val/loss: 0.8784508109092712
val/mlm_loss: 1.7663766145706177


Validation: |                                                                                                 …

val/accuracy: 0.7977250814437866
val/f1: 0.7979185581207275
val/taskclf_loss: 0.6227229237556458
val/loss: 0.8778030276298523
val/mlm_loss: 1.7642667293548584


Validation: |                                                                                                 …

`Trainer.fit` stopped: `max_epochs=10` reached.


val/accuracy: 0.7976053357124329
val/f1: 0.7977988123893738
val/taskclf_loss: 0.6225208044052124
val/loss: 0.880003809928894
val/mlm_loss: 1.7748185396194458


Best checkpoint path: checkpoints/lightning_logs/version_2/checkpoints/task-TESUni-epoch=02-val_loss=0.82.ckpt
Saved epoch checkpoint path: checkpoints/lightning_logs/version_2/checkpoints/TESUni-epoch=05.ckpt


LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]


Source genre: telephone
Target genre: slate
Number of target samples: 69575


Testing: |                                                                                                    …

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Test metric        ┃       DataLoader 0        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│   source_test/accuracy    │    0.8021312952041626     │
│      source_test/f1       │    0.8014911413192749     │
│   source_test/f1_macro    │    0.7925439476966858     │
│     source_test/loss      │    0.6297109127044678     │
│   target_test/accuracy    │    0.7413594722747803     │
│      target_test/f1       │    0.7409778237342834     │
│   target_test/f1_macro    │    0.7321345210075378     │
│     target_test/loss      │    0.8007987141609192     │
└───────────────────────────┴───────────────────────────┘

Test Results Last Epoch: [{'source_test/loss': 0.6297109127044678, 'source_test/accuracy': 0.8021312952041626, 'source_test/f1': 0.8014911413192749, 'source_test/f1_macro': 0.7925439476966858, 'target_test/loss': 0.8007987141609192, 'target_test/accuracy': 0.7413594722747803, 'target_test/f1': 0.7409778237342834, 'target_test/f1_macro': 0.7321345210075378}]
Best checkpoint path: checkpoints/lightning_logs/version_2/checkpoints/task-TESUni-epoch=02-val_loss=0.82.ckpt
Saved epoch checkpoint path: checkpoints/lightning_logs/version_2/checkpoints/TESUni-epoch=05.ckpt


Some weights of BertAdapterModel were not initialized from the model checkpoint at bert-base-uncased and are newly initialized: ['heads.default.3.bias']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.


LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]


Testing: |                                                                                                    …

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Test metric        ┃       DataLoader 0        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│   source_test/accuracy    │    0.7959389090538025     │
│      source_test/f1       │    0.7954172492027283     │
│   source_test/f1_macro    │    0.7852252721786499     │
│     source_test/loss      │     0.533909797668457     │
│   target_test/accuracy    │    0.7314227819442749     │
│      target_test/f1       │    0.7306737303733826     │
│   target_test/f1_macro    │    0.7207374572753906     │
│     target_test/loss      │    0.6879998445510864     │
└───────────────────────────┴───────────────────────────┘

Test Results on Best Model: [{'source_test/loss': 0.533909797668457, 'source_test/accuracy': 0.7959389090538025, 'source_test/f1': 0.7954172492027283, 'source_test/f1_macro': 0.7852252721786499, 'target_test/loss': 0.6879998445510864, 'target_test/accuracy': 0.7314227819442749, 'target_test/f1': 0.7306737303733826, 'target_test/f1_macro': 0.7207374572753906}]


Some weights of BertAdapterModel were not initialized from the model checkpoint at bert-base-uncased and are newly initialized: ['heads.default.3.bias']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.


LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]


Testing: |                                                                                                    …

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Test metric        ┃       DataLoader 0        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│   source_test/accuracy    │    0.8026353716850281     │
│      source_test/f1       │    0.8020183444023132     │
│   source_test/f1_macro    │    0.7930124402046204     │
│     source_test/loss      │    0.6315090656280518     │
│   target_test/accuracy    │    0.7423675060272217     │
│      target_test/f1       │    0.7419013977050781     │
│   target_test/f1_macro    │    0.7330108880996704     │
│     target_test/loss      │    0.8035094141960144     │
└───────────────────────────┴───────────────────────────┘

Test Results on saved epoch: [{'source_test/loss': 0.6315090656280518, 'source_test/accuracy': 0.8026353716850281, 'source_test/f1': 0.8020183444023132, 'source_test/f1_macro': 0.7930124402046204, 'target_test/loss': 0.8035094141960144, 'target_test/accuracy': 0.7423675060272217, 'target_test/f1': 0.7419013977050781, 'target_test/f1_macro': 0.7330108880996704}]


In [6]:
results.items()

dict_items([('last_epoch', {'source_test/loss': [0.573732852935791, 0.5562842488288879, 0.6297109127044678], 'source_test/accuracy': [0.8117079138755798, 0.822796642780304, 0.8021312952041626], 'source_test/f1': [0.8116697072982788, 0.8225594162940979, 0.8014911413192749], 'source_test/f1_macro': [0.8027613162994385, 0.8125101327896118, 0.7925439476966858], 'target_test/loss': [0.759901762008667, 0.7413967847824097, 0.8007987141609192], 'target_test/accuracy': [0.7332949042320251, 0.7282546162605286, 0.7413594722747803], 'target_test/f1': [0.7329106330871582, 0.7278156876564026, 0.7409778237342834], 'target_test/f1_macro': [0.723328709602356, 0.717542290687561, 0.7321345210075378]}), ('best_model', {'source_test/loss': [0.5176754593849182, 0.5136334896087646, 0.533909797668457], 'source_test/accuracy': [0.7986031174659729, 0.7967309951782227, 0.7959389090538025], 'source_test/f1': [0.7992354035377502, 0.7970972061157227, 0.7954172492027283], 'source_test/f1_macro': [0.7878866791725159,

| Scenario    | Metric                 | Mean      | Std       |
|-------------|------------------------|-----------|-----------|
| Last Epoch  | Source Test Loss        | 0.58691   | 0.03037   |
|             | Source Test Accuracy    | 0.81221   | 0.01027   |
|             | Source Test F1          | 0.81191   | 0.01053   |
|             | Source Test F1 Macro    | 0.80261   | 0.01002   |
|             | Target Test Loss        | 0.76737   | 0.03022   |
|             | Target Test Accuracy    | 0.73430   | 0.00654   |
|             | Target Test F1          | 0.73390   | 0.00657   |
|             | Target Test F1 Macro    | 0.72434   | 0.00741   |
| Best Model  | Source Test Loss        | 0.52107   | 0.00934   |
|             | Source Test Accuracy    | 0.79709   | 0.00121   |
|             | Source Test F1          | 0.79792   | 0.00160   |
|             | Source Test F1 Macro    | 0.78674   | 0.00132   |
|             | Target Test Loss        | 0.69406   | 0.01347   |
|             | Target Test Accuracy    | 0.71722   | 0.01282   |
|             | Target Test F1          | 0.71688   | 0.01291   |
|             | Target Test F1 Macro    | 0.70870   | 0.01156   |
| Epoch Saved | Source Test Loss        | 0.58756   | 0.03883   |
|             | Source Test Accuracy    | 0.81272   | 0.01011   |
|             | Source Test F1          | 0.81243   | 0.01052   |
|             | Source Test F1 Macro    | 0.80377   | 0.01017   |
|             | Target Test Loss        | 0.76816   | 0.03213   |
|             | Target Test Accuracy    | 0.73464   | 0.00703   |
|             | Target Test F1          | 0.73420   | 0.00702   |
|             | Target Test F1 Macro    | 0.72456   | 0.00757   |


In [7]:
# Calculate mean and standard deviation for each scenario
mean_results = {scenario: {key: np.mean(values) for key, values in metrics.items()} for scenario, metrics in results.items()}
std_results = {scenario: {key: np.std(values) for key, values in metrics.items()} for scenario, metrics in results.items()}

# Log mean and standard deviation results to wandb
wandb.init(project=project_name, name=f'{domain}_mean_results')
for scenario in mean_results:
    for key, value in mean_results[scenario].items():
        wandb.log({f"{scenario}/{key}": value})
        wandb.log({f"{scenario}/{key}_std": std_results[scenario][key]})
wandb.finish()

print("Mean Results:", mean_results)
print("Standard Deviation Results:", std_results)

# # Save the best model's adapter
# if model:
#     adapter_save_path = f"../../saved/adapter_after_run/{hparams['task_adapter_name']}"
#     model.save_adapter(adapter_save_path, hparams['task_adapter_name'])
#     print(f"Adapter saved to {adapter_save_path}")
# else:
#     print("No best model to save.")

wandb: wandb version 0.17.8 is available!  To upgrade, please run:
wandb:  $ pip install wandb --upgrade


wandb: Tracking run with wandb version 0.17.0


wandb: Run data is saved locally in /home/guest/Desktop/projects/third-experiments/domain_adaptation_project/mixed/mixed/wandb/run-20240901_035357-eajcc972
wandb: Run `wandb offline` to turn off syncing.


wandb: Syncing run TES_mean_results


wandb: ⭐️ View project at https://wandb.ai/mrawhani/mixed_edited


wandb: 🚀 View run at https://wandb.ai/mrawhani/mixed_edited/runs/eajcc972


wandb:                                                                                


wandb: 
wandb: Run history:
wandb:      best_model/source_test/accuracy ▁
wandb:  best_model/source_test/accuracy_std ▁
wandb:            best_model/source_test/f1 ▁
wandb:      best_model/source_test/f1_macro ▁
wandb:  best_model/source_test/f1_macro_std ▁
wandb:        best_model/source_test/f1_std ▁
wandb:          best_model/source_test/loss ▁
wandb:      best_model/source_test/loss_std ▁
wandb:      best_model/target_test/accuracy ▁
wandb:  best_model/target_test/accuracy_std ▁
wandb:            best_model/target_test/f1 ▁
wandb:      best_model/target_test/f1_macro ▁
wandb:  best_model/target_test/f1_macro_std ▁
wandb:        best_model/target_test/f1_std ▁
wandb:          best_model/target_test/loss ▁
wandb:      best_model/target_test/loss_std ▁
wandb:     epoch_saved/source_test/accuracy ▁
wandb: epoch_saved/source_test/accuracy_std ▁
wandb:           epoch_saved/source_test/f1 ▁
wandb:     epoch_saved/source_test/f1_macro ▁
wandb: epoch_saved/source_test/f1_macro_std ▁
wandb:

wandb: 🚀 View run TES_mean_results at: https://wandb.ai/mrawhani/mixed_edited/runs/eajcc972
wandb: ⭐️ View project at: https://wandb.ai/mrawhani/mixed_edited
wandb: Synced 4 W&B file(s), 0 media file(s), 0 artifact file(s) and 0 other file(s)


wandb: Find logs at: ./wandb/run-20240901_035357-eajcc972/logs


Mean Results: {'last_epoch': {'source_test/loss': 0.586576004823049, 'source_test/accuracy': 0.8122119506200155, 'source_test/f1': 0.8119067549705505, 'source_test/f1_macro': 0.802605132261912, 'target_test/loss': 0.7673657536506653, 'target_test/accuracy': 0.7343029975891113, 'target_test/f1': 0.7339013814926147, 'target_test/f1_macro': 0.7243351737658182}, 'best_model': {'source_test/loss': 0.5217395822207133, 'source_test/accuracy': 0.797091007232666, 'source_test/f1': 0.797249952952067, 'source_test/f1_macro': 0.7867420117060343, 'target_test/loss': 0.6947266856829325, 'target_test/accuracy': 0.7178859313329061, 'target_test/f1': 0.7168845534324646, 'target_test/f1_macro': 0.7086983720461527}, 'epoch_saved': {'source_test/loss': 0.5875607331593832, 'source_test/accuracy': 0.812716007232666, 'source_test/f1': 0.812424898147583, 'source_test/f1_macro': 0.8031044602394104, 'target_test/loss': 0.7688279946645101, 'target_test/accuracy': 0.7346390088399252, 'target_test/f1': 0.734201133

In [8]:
print('dones')

dones


In [9]:
best_val_loss

inf